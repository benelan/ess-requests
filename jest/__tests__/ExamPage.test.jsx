import React from 'react'
import { render, screen, fireEvent } from '@testing-library/react'
import renderer from 'react-test-renderer'
import ExamPage from '../../pages/exam'

const { act } = renderer

describe('<Exam /> snapshot', () => {
  it('matches component', () => {
    const examTree = renderer.create(<ExamPage />).toJSON()
    expect(examTree).toMatchSnapshot()
  })
})

describe('<Exam /> renders', () => {
  it('user name', async () => {
    render(<ExamPage />)
    expect(await screen.findByRole('textbox', { name: 'Employee Name' })).toHaveValue('Test User')
  })
})

describe('<Exam /> events', () => {
  it('bootstrap validation on submit', async () => {
    render(<ExamPage />)
    await act(async () => screen.findByRole('textbox', { name: 'Employee Name' }))
    const examForm = screen.getByRole('form', { name: /Exam Request Form/ })
    expect(examForm).not.toHaveClass('was-validated')
    fireEvent.click(screen.getByRole('button', { name: 'Submit' }))
    expect(examForm).toHaveClass('was-validated')
  })

  // it('submit filled in form', async () => {
  //   render(<Exam />)
  //   await act(async () => screen.findByRole('textbox', { name: 'Employee Name' }))
  //   const examForm = screen.getByRole('form', { name: /Exam Request Form/ })
  //   expect(examForm).not.toHaveClass('was-validated')
  //   fireEvent.click(screen.getByRole('button', { name: 'Submit' }))
  //   expect(submitForm).not.toHaveBeenCalled()
  //   expect(examForm).toHaveClass('was-validated')
  // })
})
