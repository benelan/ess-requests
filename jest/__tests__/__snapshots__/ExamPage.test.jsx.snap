// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Exam /> snapshot matches component 1`] = `
Array [
  <title>
    Exam Request
  </title>,
  <nav
    className="navbar navbar-expand navbar-light border-bottom"
    style={
      Object {
        "background": "white",
      }
    }
  >
    <ul
      className="navbar-nav"
    >
      <li
        aria-label="Home Link"
        className="nav-item"
      >
        <a
          className="nav-link"
          href="/"
          onClick={[Function]}
          onMouseEnter={[Function]}
        >
          Home
        </a>
      </li>
      <li
        aria-label="Training Link"
        className="nav-item"
      >
        <a
          className="nav-link"
          href="/training"
          onClick={[Function]}
          onMouseEnter={[Function]}
        >
          Training
        </a>
      </li>
      <li
        aria-label="Exam Link"
        className="nav-item active"
      >
        <a
          className="nav-link"
          href="/exam"
          onClick={[Function]}
          onMouseEnter={[Function]}
        >
          Exam
        </a>
      </li>
    </ul>
  </nav>,
  <h3
    className="text-center m-4"
  >
    Request for Exam Certification
  </h3>,
  <div
    className="container"
  >
    <form
      aria-label="Exam Request Form"
      className="needs-validation"
      noValidate={true}
      onSubmit={[Function]}
    >
      <div
        className="form-row mt-3"
      >
        <div
          className="form-group col-md-3"
        >
          <label
            htmlFor="nameEmployee"
          >
            Employee Name
          </label>
          <input
            className="form-control"
            disabled={true}
            id="nameEmployee"
            required={true}
            type="text"
            value=""
          />
        </div>
        <div
          className="form-group col-md-3"
        >
          <label
            htmlFor="employeeNumber"
          >
            Employee Number
          </label>
          <input
            className="form-control"
            id="employeeNumber"
            onChange={[Function]}
            required={true}
            type="number"
          />
        </div>
        <div
          className="form-group col-md-3"
        >
          <label
            htmlFor="employeeUnit"
          >
            Employee Unit
          </label>
          <select
            className="form-control"
            id="employeeUnit"
            onChange={[Function]}
          >
            <option
              value="Supt-ArcGIS-Unit-Mgmt@esri.com"
            >
              Online
            </option>
            <option
              value="Supt-Enterprise-Unit-Mgmt@esri.com"
            >
              Enterprise
            </option>
            <option
              value="Supt-Desktop-Unit-Mgmt@esri.com"
            >
              Desktop
            </option>
            <option
              value="Supt-DaDP-Unit-Mgmt@esri.co"
            >
              DaDP Product
            </option>
            <option
              value="Supt-NORUS-Unit-Mgmt@esri.com"
            >
              NORUS
            </option>
            <option
              value="SWhittington@esri.com"
            >
              TAMS
            </option>
            <option
              value="Christian_Wells@esri.com"
            >
              Readiness
            </option>
          </select>
        </div>
        <div
          className="form-group col-md-3"
        >
          <label
            htmlFor="employeeLocation"
          >
            Employee Location
          </label>
          <select
            className="form-control"
            id="employeeLocation"
            onChange={[Function]}
          >
            <option>
              Redlands
            </option>
            <option>
              Charlotte
            </option>
            <option>
              Washington DC
            </option>
            <option>
              St Louis
            </option>
          </select>
        </div>
      </div>
      <div
        className="form-row mt-3"
      >
        <div
          className="form-group col-md-5"
        >
          <label
            htmlFor="examName"
          >
            Exam Name
          </label>
          <input
            className="form-control"
            id="examName"
            onChange={[Function]}
            required={true}
            type="text"
          />
        </div>
        <div
          className="form-group col-md-4"
        >
          <label
            htmlFor="vendor"
          >
            Vendor
          </label>
          <input
            className="form-control"
            id="vendor"
            onChange={[Function]}
            required={true}
            type="text"
          />
        </div>
        <div
          className="form-group col-md-3"
        >
          <label
            htmlFor="examLocation"
          >
            Exam Location
          </label>
          <select
            className="form-control"
            id="examLocation"
            onChange={[Function]}
          >
            <option>
              Redlands
            </option>
            <option>
              Charlotte
            </option>
            <option>
              Washington DC
            </option>
            <option>
              St Louis
            </option>
            <option>
              Remote
            </option>
          </select>
        </div>
      </div>
      <div
        className="form-row mt-3"
      >
        <div
          className="form-group col-md-2"
        >
          <label
            htmlFor="cost"
          >
            Cost
          </label>
          <div
            className="input-group mb-3"
          >
            <div
              className="input-group-prepend"
            >
              <span
                className="input-group-text"
              >
                $
              </span>
            </div>
            <input
              className="form-control"
              id="cost"
              onChange={[Function]}
              required={true}
              type="number"
            />
          </div>
        </div>
        <div
          className="form-group col-md-10"
        >
          <label
            htmlFor="justification"
          >
            Justification
          </label>
          <input
            className="form-control"
            id="justification"
            onChange={[Function]}
            required={true}
          />
        </div>
      </div>
      <div
        className="form-row mt-3"
      >
        <div
          className="form-group col"
        >
          <input
            className="btn btn-primary"
            type="submit"
            value="Submit"
          />
        </div>
      </div>
    </form>
  </div>,
]
`;
